action/protocol/staking/protocol.go

```diff
@@ -492,7 +538,11 @@ func (p *Protocol) ActiveCandidates(ctx context.Context, sr protocol.StateReader
 			}
 			list[i].Votes.Add(list[i].Votes, contractVotes)
 		}
-		if list[i].SelfStake.Cmp(p.config.RegistrationConsts.MinSelfStake) >= 0 {
+		active, err := p.isActiveCandidate(ctx, c, list[i])
```

Comment:
why are unactive cands included in L527?

Reply:
There are two scenarios: 
- The self-stake bucket has been unstaked. 
- The endorsement has expired. 

Both situations belong to a registered delegate, but they cannot participate in consensus due to a lack of staking.

Reply:
ok. It seems the state of candidates aren't tracked in the db(active/unactive). Pls add ur comments to explain why the list is filtered above L541